Fix host name when syslog is used

thiagoftsm opened the pull request:
##### Summary
Fixes #10245 

This PR changes the variable used to reported host inside syslog, previously we were having logs like this:

```
Nov 23 19:56:41 hades thiago: netdata CRITICAL on  at Mon Nov 23 19:56:40 UTC 2020: system.cpu 4.23%
Nov 23 19:56:59 hades thiago: netdata CRITICAL on  at Mon Nov 23 19:56:58 UTC 2020: system.cpu 7.12%
Nov 23 19:56:59 hades thiago: netdata CRITICAL on  at Mon Nov 23 19:56:58 UTC 2020: system.cpu 18.3%
```

After the fix we start to have the expected logs for `parent` and `child`:

```
Nov 23 20:13:26 hades thiago: netdata CRITICAL on zeus at Mon Nov 23 19:55:09 UTC 2020: mem.available 4%
Nov 23 20:13:33 hades thiago: netdata CRITICAL on zeus at Mon Nov 23 20:13:32 UTC 2020: system.cpu 4.78%
Nov 23 20:13:39 hades thiago: netdata CRITICAL on hades at Mon Nov 23 20:13:32 UTC 2020: system.cpu 4.82%
Nov 23 20:14:02 hades thiago: netdata CRITICAL on hades at Mon Nov 23 20:14:01 UTC 2020: system.cpu 4.56%

```

##### Component Name
Health
##### Test Plan
### Simple scenario

I observed that the problem was bigger than reported, so there is another way to test this PR:

1 - Compile this PR on `parent` and set your `health_alarm_notify.conf` with the following variables:

```
SEND_SYSLOG="YES"
SYSLOG_FACILITY='local6'
DEFAULT_RECIPIENT_SYSLOG="netdata" 
```

2 - Go to `/usr/libexec/netdata/plugins.d/` and run:

```
$ ./alarm-notify.sh test
```

And check your syslog messages.

### Scenario reported per user

1 - Compile this PR on `parent` and set your `health_alarm_notify.conf` with the following variables:

```
SEND_SYSLOG="YES"
SYSLOG_FACILITY='local6'
DEFAULT_RECIPIENT_SYSLOG="netdata" 
```

2 - Write an alarm that will be easily raised, for example :

```
   alarm: example_alarm
      on: example.random
   every: 2s
  lookup: sum -1s at 0 every 1 percentage foreach *
    warn: $this > (($status >= $WARNING)  ? (55) : (75))
    crit: $this > (($status == $CRITICAL) ? (75) : (95))
    info: random test
      to: sysadmin
``` 
3 - Enable python `example` module

4 - Start `parent`
5 - You can compile any branch at `child`, because the problem is happening on `parent`.
6 - Also enable python `example` module at slave
7 - Start Netdata with `health` disabled at child.
8 - Take a look at your syslog messages to confirm the `host` is present at parent and child messages.
##### Additional Information

ilyam8 commented:
@thiagoftsm could you calrify difference between `{host}` and `{args_host}`.

ilyam8 commented:
I see if `args_host` is not empty, then we set `host` to its value.

https://github.com/netdata/netdata/blob/4de1c2e6cca67d8fcd87ebd88879293042aeda8c/health/notifications/alarm-notify.sh.in#L244-L250

---

Then how come `host` is empty when `args_host` is not?

thiagoftsm commented:
> I see if `args_host` is not empty, then we set `host` to its value.
> 
> https://github.com/netdata/netdata/blob/4de1c2e6cca67d8fcd87ebd88879293042aeda8c/health/notifications/alarm-notify.sh.in#L244-L250
> 
> Then how come `host` is empty when `args_host` is not?

You are right @ilyam8 , but on my environment and also on our user environment, we were not having the variables [prefix](https://github.com/netdata/netdata/blob/4de1c2e6cca67d8fcd87ebd88879293042aeda8c/health/notifications/alarm-notify.sh.in#L1856) and [temp1](https://github.com/netdata/netdata/blob/4de1c2e6cca67d8fcd87ebd88879293042aeda8c/health/notifications/alarm-notify.sh.in#L1857) with different values, so `alarm-notify.sh` was using a clean `host` variable. The [args_host](https://github.com/netdata/netdata/blob/4de1c2e6cca67d8fcd87ebd88879293042aeda8c/health/notifications/alarm-notify.sh.in#L214) on the other hand, is given for the script as an argument. This is the motive I decided to use it, instead the local variable `$host`. I am bringing a fix that is working with the safest approach to have the host name.

ilyam8 commented:
Ok, `send_syslog` modifies `host` global variable, that is the problem. Shouldnt be using it.

thiagoftsm commented:
> Ok, `send_syslog` modifies `host` global variable, that is the problem. Shouldnt be using it.

I changed the variable name and I also changed the conditions to fix the reported problem.

ilyam8 commented on health/notifications/alarm-notify.sh.in:
```diff
@@ -1859,36 +1859,40 @@ send_syslog() {
+      server="${args_host}"
+    fi
+
+    message="${prefix} ${status} on ${server} at ${date}: ${chart} ${value_string}"
```
```suggestion
    message="${prefix} ${status} on ${host} at ${date}: ${chart} ${value_string}"
```

ilyam8 replied:
`server` is a destination we want to send log message to, right?

thiagoftsm replied:
good catch, thank you ! I changed all variables, but this should be `host`.

ilyam8 commented on health/notifications/alarm-notify.sh.in:
```diff
@@ -1859,36 +1859,40 @@ send_syslog() {
 
+    if [ -z ${server} ] ; then
+      server="${args_host}"
+    fi
```
why we need this?

thiagoftsm replied:
We needed this when we were using `server` to write the `message` , but now that we are using `host`, this became dead code. It was removed! thanks!

ilyam8 commented on health/notifications/alarm-notify.sh.in:
```diff
@@ -1887,8 +1887,8 @@ send_syslog() {
 
-    if [ ${host} ]; then
-      logger_options="${logger_options} -n ${host}"
+    if [ ${server} ] && [ -z $temp1 ]; then
```
Why do we need `&& [ -z $temp1 ]`? I this if there is `${server}` then we need to send the message to the remote host, isn't `${server}` enough?

To be honest i don't really follow all this magic with temp1/temp2/etc 😅

thiagoftsm replied:
the `tempX` variable is a little bit confused, we will need another PR to rename them. 

The comparison I added it is not more necessary after to remove the conditional with the previous commit. I removed it.
